Raven.Database/Raft/ClusterManagementHttpClient.cs

```diff
@@ -345,7 +345,20 @@ public Task SendNodeUpdateAsync(NodeConnectionInfo node)
                 }
                 else
                 {
+                    // before we send remove from cluster wait until configuration will be propaged
```

Comment:
In general we have to wait until term is changed because in other case we get : InvalidOperationException. So alternative is to spin until logEntry.Term == PersistentState.CurrentTerm. 

For similar reason I've put second task delay. In raftEngine.RemoveFromCluster we have new topology on leader and in meanwhile we send new topology to target nodes (via append entries). In append entries we check whether messages comes from proper topology and before changing them we received SendInitializeNewClusterForAsync message which changed topology id from topo1 -> topo2, after doing this append entries was finishing and reverting topology back to topo1. As result no change was made.